src/pages/RecipePage.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from "react";
+import { useHistory } from "react-router-dom";
+import styled from "styled-components";
+import axios from "axios";
+import { useParams } from "react-router-dom";
+import { useQuery } from "../utils";
+
+const Title = styled.h1`
+  color: #4f46e5;
+  margin-bottom: 1rem;
+`;
+
+const Description = styled.p`
+  font-size: 1.05rem;
+  font-weight: 400;
+  margin-bottom: 2rem;
+`;
+
+const IngredientsHeading = styled.div`
+  font-size: 1.15rem;
+  font-weight: 700;
+`;
+
+const IngredientsList = styled.ul`
+  line-height: 1.5rem;
+`;
+
+const BackButton = styled.button`
+  background-color: #a5b4fc;
+  padding: 0.6rem;
+  color: white;
+  font-weight: 700;
+  font-size: 1rem;
+  border: none;
+  border-radius: 0.5rem;
+  cursor: pointer;
+
+  &:hover {
+    background-color: #818cf8;
+  }
+`;
+
+const EditButton = styled.button`
+  background-color: #a5b4fc;
+  margin-left: 0.5rem;
+  padding: 0.6rem;
+  color: white;
+  font-weight: 700;
+  font-size: 1rem;
+  border: none;
+  border-radius: 0.5rem;
+  cursor: pointer;
+
+  &:hover {
+    background-color: #818cf8;
+  }
+`;
+
+const DeleteButton = styled.button`
+  background-color: #fda4af;
+  margin-left: 0.5rem;
+  padding: 0.6rem;
+  color: #9f1239;
+  font-weight: 700;
+  font-size: 1rem;
+  border: none;
+  border-radius: 0.5rem;
+  cursor: pointer;
+
+  &:hover {
+    background-color: #fb7185;
+  }
+`;
+
+const Alert = styled.div`
+  background-color: #f8d7da;
+  color: #721c24;
+  padding: 1rem;
+  border-radius: 0.5rem;
+  margin-bottom: 1rem;
+  margin-top: 1rem;
+`;
+
+export default function RecipePage() {
+  const { recipeId } = useParams();
+  const { updated } = useQuery();
+  const history = useHistory();
+  const [recipe, setRecipe] = useState(null);
+
+  useEffect(() => {
+    axios.get(`/api/recipes/${recipeId}/`).then((response) => {
+      setRecipe(response.data);
+    });
+  }, [recipeId]);
+
+  const goToHome = () => history.push("/");
+  const goToEditPage = () => history.push(`/recipes/${recipeId}/edit`);
+  const deleteRecipe = () =>
+    axios
+      .delete(`/api/recipes/${recipeId}/`)
+      .then((response) => {
+        if (response.status === 204) history.push("/?deleted=true");
+      })
+      .catch((err) => console.error(err));
+
+  return (
+    <div>
+      {recipe && (
```

Comment:
I think this component should only render the recipe once you get the data. 

There should be the API call at the higher level and if it succeeds it gets the data and pass down to this component. Otherwise, if it fails to find a recipe it can render something else.

It makes it easier to test!

Reply:
sure, it makes sense to extract a Recipe component 👍

Reply:
+1